File: la-gula-pizzeria/src/app/shared/services/customer/customer-service.ts
import { Injectable } from '@angular/core';
import CustomerResponse from '../../model/customer/response/customerResponse.model';
import { Observable } from 'rxjs';
import { GATEWAY_URL, SERVICES_PATHS } from '../config/config';
import CustomerRequest from '../../model/customer/request/customerRequest.model';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {
  
  path = `${GATEWAY_URL}${SERVICES_PATHS.CUSTOMER}` + '/customers';

  constructor(private http: HttpClient) { }

  getAllCustomers() : Observable<CustomerResponse[]> {
    return this.http.get<CustomerResponse[]>(this.path);
  }
  getCustomerById(id: number) : Observable<CustomerResponse> {
    return this.http.get<CustomerResponse>(`${this.path}/${id}`);
  }
  saveCustomer(customer: CustomerRequest) : Observable<CustomerResponse> {
    return this.http.post<CustomerResponse>(this.path, customer);
  }
  updateCustomer(id: number, customer: CustomerRequest) : Observable<CustomerResponse> {
    return this.http.put<CustomerResponse>(`${this.path}/${id}`, customer);
  }
  deleteCustomer(id: number) : Observable<void> {
    return this.http.delete<void>(`${this.path}/${id}`);
  }
}

File: back end/customer-service/src/main/java/com/cibertec/service/CustomerService.java
package com.cibertec.service;

import com.cibertec.dto.CustomerRequest;
import com.cibertec.dto.CustomerResponse;

import java.util.List;

public interface CustomerService {
    List<CustomerResponse> getAllCustomers();

    CustomerResponse getCustomerById(Long id);

    CustomerResponse createCustomer(CustomerRequest CustomerRequest);

    CustomerResponse updateCustomer(Long id, CustomerRequest CustomerRequest);

    void deleteCustomer(Long id);

    List<CustomerResponse> getCustomersById(Long id);

    CustomerResponse getCustomerByUserId(Long id);
}
